# 範例

取得 SIFT 特徵

- 轉成灰階圖片
- 需要額外安裝 OpenCV 相關套件

In [1]:
import numpy as np
import cv2

img = cv2.imread('data/lena.png')

In [2]:
# 轉灰階圖片
img_gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)

# 建立 SIFT 物件
SIFT_detector = cv2.xfeatures2d.SIFT_create()

# 取得 SIFT 關鍵點位置
keypoints = SIFT_detector.detect(img_gray, None)

#　畫圖 + 顯示圖片
img_show = cv2.drawKeypoints(img_gray, keypoints, img)
while True:
    cv2.imshow('SIFT', img_show)
    k = cv2.waitKey(0)
    if k == 27:
        cv2.destroyAllWindows()
        break

AttributeError: module 'cv2.cv2' has no attribute 'xfeatures2d'

In [3]:
!pip list

Package               Version            
--------------------- -------------------
-pencv-python         4.1.1.26           
attrs                 19.3.0             
backcall              0.1.0              
bleach                3.1.0              
certifi               2019.9.11          
colorama              0.4.1              
decorator             4.4.1              
defusedxml            0.6.0              
entrypoints           0.3                
importlib-metadata    0.23               
ipykernel             5.1.3              
ipython               7.9.0              
ipython-genutils      0.2.0              
jedi                  0.15.1             
Jinja2                2.10.3             
jsonschema            3.1.1              
jupyter-client        5.3.4              
jupyter-core          4.6.1              
MarkupSafe            1.1.1              
mistune               0.8.4              
more-itertools        7.2.0              
nbconvert             5.6.1       

Package               Version            
--------------------- -------------------
-pencv-python         4.1.1.26           
attrs                 19.3.0             
backcall              0.1.0              
bleach                3.1.0              
certifi               2019.9.11          
colorama              0.4.1              
decorator             4.4.1              
defusedxml            0.6.0              
entrypoints           0.3                
importlib-metadata    0.23               
ipykernel             5.1.3              
ipython               7.9.0              
ipython-genutils      0.2.0              
jedi                  0.15.1             
Jinja2                2.10.3             
jsonschema            3.1.1              
jupyter-client        5.3.4              
jupyter-core          4.6.1              
MarkupSafe            1.1.1              
mistune               0.8.4              
more-itertools        7.2.0              
nbconvert             5.6.1              
nbformat              4.4.0              
notebook              6.0.2              
numpy                 1.17.4             
opencv-contrib-python 3.4.2.16           
opencv-python         3.4.2.16           
pandocfilters         1.4.2              
parso                 0.5.1              
pickleshare           0.7.5              
pip                   19.3.1             
prometheus-client     0.7.1              
prompt-toolkit        2.0.10             
Pygments              2.4.2              
pyrsistent            0.15.5             
python-dateutil       2.8.1              
pywin32               223                
pywinpty              0.5.5              
pyzmq                 18.1.0             
Send2Trash            1.5.0              
setuptools            41.6.0.post20191030
six                   1.13.0             
terminado             0.8.2              
testpath              0.4.4              
tornado               6.0.3              
traitlets             4.3.3              
wcwidth               0.1.7              
webencodings          0.5.1              
wheel                 0.33.6             
wincertstore          0.2                
zipp                  0.6.0  